DATA/production/configurations/2022/LHC22f/apass1/async_pass.sh

```diff
@@ -589,70 +605,114 @@ if [[ $ALIEN_JDL_AODOFF != 1 ]]; then
 	echo "exit code from AO2D check is " $exitcode
 	exit $exitcode
       fi
-      if [[ -z $ALIEN_JDL_DONOTMERGEAODS ]] || [[ $ALIEN_JDL_DONOTMERGEAODS == 0 ]]; then
-	ls AO2D.root > list.list
-	timeStartMerge=`date +%s`
-	time o2-aod-merger --input list.list --output AO2D_merged.root
-	timeEndMerge=`date +%s`
-	timeUsedMerge=$(( $timeUsedMerge+$timeEndMerge-$timeStartMerge ))
-	exitcode=$?
-	echo "exitcode = $exitcode"
-	if [[ $exitcode -ne 0 ]]; then
-	  echo "exit code from aod-merger to merge DFs is " $exitcode > validation_error.message
-	  echo "exit code from aod-merger to merge DFs is " $exitcode
-	  exit $exitcode
-	fi
-	echo "Checking AO2Ds with merged DFs"
-	timeStartCheckMergedAOD=`date +%s`
-	time root -l -b -q '$O2DPG_ROOT/DATA/production/common/readAO2Ds.C("AO2D_merged.root")' > checkAO2D_merged.log
-	timeEndCheckMergedAOD=`date +%s`
-	timeUsedCheckMergedAOD=$(( $timeUsedCheckMergedAOD+$timeEndCheckMergedAOD-$timeStartCheckMergedAOD ))
-	exitcode=$?
-	if [[ $exitcode -ne 0 ]]; then
-	  echo "exit code from AO2D with merged DFs check is " $exitcode > validation_error.message
-	  echo "exit code from AO2D with merged DFs check is " $exitcode
-	  echo "We will keep the AO2Ds with unmerged DFs"
-	else
-	  echo "All ok, replacing initial AO2D.root file with the one with merged DFs"
-	  mv AO2D_merged.root AO2D.root
-	fi
+    fi
+    cd -
+    ls $AOD_FILE > list_$AOD_DIR.list
+    echo "$AOD_DIR" >> $JOB_LIST
+  done
+  if [[ -z $ALIEN_JDL_DONOTMERGEAODS ]] || [[ $ALIEN_JDL_DONOTMERGEAODS == 0 ]]; then
+    # spawning the parallel merging
+    timeStartMerge=`date +%s`
+    arr=()
+    aods=()
+    mergedok=()
+    i=0
+    while IFS= read -r line; do
+      while [[ $CURRENT_POOL_SIZE -ge $MAX_POOL_SIZE ]]; do
+	CURRENT_POOL_SIZE=`jobs -r | wc -l`
```

Comment:
you should add a sleep here, otherwise we have one processes spinning checking the variable

Reply:
Ok, I will. It will decrease the performance, though, no?

Reply:
`sleep 1` added.

Reply:
> Ok, I will. It will decrease the performance, though, no?

It will increase it. Without the sleep, one cpu is busy with the script...